docs/CODING_GUIDELINES.md

```diff
@@ -0,0 +1,304 @@
+    ```
+  - explain unclear code in comments, rather refactor the code to make it clear.
+- Do ✅
+  - use comments explain why this code is necessary.
+  - explain unidiomatic code in comments.
+  - provide links to the original source of copied code.
+  - include links to external references where they will be most helpful.
+  - add comments when fixing bugs which explains the edge case that caused the bug.
+  - use comments to mark incomplete implementations.
+
+### Conditional values
+
+- Prefer using `nullish coalescing operator` over of `ternary` or `&& / ||` operators.
+
+  ```ts
+  // ❌
+  const displayName = user.firstName ? user.firstName : user.lastName;
+  ```
+
+  ```ts
+  // ❌
+  const displayName = user.firstName || user.lastName;
+  ```
+
+  ```ts
+  // ✅
+  const displayName = user.firstName ?? user.lastName;
+  ```
+
+- Do not use `ternary` operator for more than two conditions
+
+  ```ts
+  // ✅
+  const connectionState = device.connected ? 'connected' : 'disconnected';
+  ```
+
+  ```ts
+  // ❌
+  const connectionState = device.connected ? 'connected' : device.isBootloader ? 'bootloader' : 'disconnected';
+  ```
+
+  ```ts
+  // ✅
+  let connectionState = 'disconnected';
+  if (device.connected) {
+      connectionState = 'connected';
+  } else if(device.isBootloader) {
+      connectionState = 'bootloader';
+  }
+  ```
+
+- Use object for mapping variables into different format.
+
+  ```ts
+  // ❌
+  let fontWeightNumber: number;
+  if(fontWeight === 'light') {
+      fontWeightNumber = 300;
+  } else if (fontWeight === 'normal') {
+      fontWeightNumber = 400;
+  } else if (fontWeight === 'bold') {
+      fontWeightNumber = 700;
+  } else {
+      fontWeightNumber = 400;
+  }
+  ```
+
+  ```ts
+  // ✅
+  const fontWeightMap = {
+      light: 300,
+      normal: 400,
+      bold: 700
+  };
+
+  const fontWeightNumber = fontWeightMap[fontWeight] ?? fontWeightMap.normal;
+  ```
+
+### React
+
+#### Styling
+
+- Use `styled-components` while creating UI components.
+- Use props on UI components to modify the styling, over directly changing the css.
+
+  ```tsx
+  // ❌
+  <Button variant="primary" style={{ border: 'none' }}>Continue</Button>
+  ```
+
+  ```tsx
+  // ✅
+  <Button variant="primary" noBoarder={true}>Continue</Button>
+  ```
+
+#### Logic
+
+- Create reusable hooks where possible.
+- Global states should be defined in Redux store.
+- Separate business logic from UI.
+
+  ```tsx
+  // ❌
+  const SetPassword = () => {
+      const handleSetPassword = (password: string) => {
+          const db = getDB();
+          const passwordHash = createHash(password);
+
+          db.setPassword(passwordHash);
+          db.encryptWithPassword(passwordHash);
+      }
+
+      return (
+        <div>
+            {/* ... */}
+        </div>
+      );
+  }
+  ```
+
+  ```tsx
+  // ✅
+  const SetPassword = () => {
+      const handleSetPassword = (password: string) => {
+          setDatabasePassword(password);
```

Comment:
Might want to add an import statement to make it clear where this function comes from